src/MongoDB.Driver/Linq/Linq3Implementation/MongoQueryProvider.cs

```
@@ -152,5 +153,14 @@ public override ExpressionTranslationOptions GetTranslationOptions()
             var database = _database ?? _collection?.Database;
             return translationOptions.AddMissingOptionsFrom(database?.Client.Settings.TranslationOptions);
         }
+
+        public override BsonDocument[] Translate<TResult>(IQueryable<TResult> queryable, out IBsonSerializer<TResult> outputSerializer)
```

Should we check if provided queryable is MongoDB queryable?

---

Or maybe receiving an expression is sufficient?

---

It would be sufficient.

I chose `IQueryable` to emphasize that it's the `Expression` associated with an `IQueryable` that is being translated, not any old expression.

But the .NET method takes an `Expression`, so we could also here.

@damieng what do you think?

---

The translate method seems to only care about the expression of the queryable so I think we should just take `Expression` here as well. We can add a note in the comments of methods to emphasize the expression should be associated with an `IQueryable` and maybe throw an exception if we catch such as a case?